package-scripts/datadog-agent/postinst

```diff
@@ -41,15 +41,20 @@ if [ "$DISTRIBUTION" != "Darwin" ]; then
               update-rc.d datadog-agent defaults
               echo "Enabling service datadog-agent"
               update-rc.d datadog-agent enable >/dev/null 2>&1
-              # Only add dd-agent user if it doesn't already exist
+              # Only create dd-agent group and/or user if they don't already exist
+              getent group dd-agent >/dev/null || echo "Creating dd-agent group" && addgroup --system dd-agent --quiet
               set +e
               id -u dd-agent >/dev/null 2>&1
               USER_EXISTS=$?
               set -e
               if [ ! $USER_EXISTS -eq 0 ]; then
                   echo "Creating dd-agent user"
-                  adduser --system dd-agent --disabled-login --shell /bin/sh --no-create-home --quiet
+                  adduser --system dd-agent --disabled-login --shell /bin/sh --no-create-home --group --quiet
                   usermod -d /opt/datadog-agent dd-agent
+              elif id -nG dd-agent | grep --invert-match --word-regexp --quiet 'dd-agent'; then
+                  # User exists but is not part of the dd-agent group
+                  echo "Adding dd-agent user to dd-agent group"
+                  usermod -g dd-agent dd-agent
```

Comment:
good point, doing this would be less aggressive if users have already set up a group for the `dd-agent` user... We'd need to filter out the `nogroup` group though.